Naggum.Compiler/FormGenerator.fs

```diff
@@ -86,18 +86,13 @@ type BodyGenerator(context : Context,
     interface IGenerator with
         member __.Generate ilGen =
             genBody ilGen body
-        member this.ReturnTypes () =
+        member __.ReturnTypes () =
             match body with
-            |[] -> [typeof<System.Void>]
-            |somelist ->
-                let tail_type = (gf.MakeGenerator context (List.rev body |> List.head)).ReturnTypes()
-                if tail_type = [typeof<System.Void>] then
-                    [typeof<obj>]
-                else tail_type
+            | [] -> [typeof<System.Void>]
+            | _ -> (gf.MakeGenerator context (List.rev body |> List.head)).ReturnTypes()
```

Comment:
Thank you, that's a good question. I'm almost sure that `List` module had no such function back the days of F# 2.0. But now it actually has! So I'll review the code and use it where possible. I'll create a separate issue for that.